homework-submission/react-week3/src/App.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from "react";
+import AddingTodoForm from "./adding-form";
+import RenderTodo from "./render-todo";
+import EditingForm from "./editing-form";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [
+        {
+          id: 1,
+          description: "Get out of bed",
+          deadline: "2017-09-11",
+          done: true
+        },
+        {
+          id: 2,
+          description: "Brush teeth",
+          deadline: "2017-09-10",
+          done: false
+        },
+        {
+          id: 3,
+          description: "Eat breakfast",
+          deadline: "2017-09-09",
+          done: false
+        }
+      ]
+    };
+    this.addItem = this.addItem.bind(this);
+    this.markAs = this.markAs.bind(this);
+    this.updateItem = this.updateItem.bind(this);
+    this.removeItem = this.removeItem.bind(this);
+    this.editMode = this.editMode.bind(this);
+    this.cancelEditMode = this.cancelEditMode.bind(this);
+
+  }
+
+  addItem(e) {
+    e.preventDefault();
+    this.state.todos.push({
+      id: Math.max(...this.state.todos.map(el => el.id)) + 1,
+      description: e.target.description.value,
+      deadline: e.target.deadline.value,
+      done: false
+    });
+    this.setState({ todos: this.state.todos });
+  }
+  removeItem(item) {
+    this.setState({ todos: this.state.todos.filter(el => el.id !== item.id) });
+  }
+  editMode(item) {
+    this.setState({ editingMode: item.id });
+  }
+  cancelEditMode() {
+    this.setState({ editingMode: false });
+  }
```

Comment:
it is not good to create a variable and then change the type of it even if the code is working
( it is boolean and then changed to number)
and you did not give a default value to this variable in the state